docs/metrics-guide/averaging-methods.md

```diff
@@ -0,0 +1,118 @@
+
+| Class | # Samples | <span title="# True Positives">TP</span> | <span title="# False Positives">FP</span> | <span title="# False Negatives">FN</span> | Precision | Recall | F1-score |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| `Airplane` | 3 | 2 | 1 | 1 | 0.67 | 0.67 | 0.67 |
+| `Boat` | 1 | 1 | 3 | 0 | 0.25 | 1.0 | 0.4 |
+| `Car` | 6 | 3 | 0 | 3 | 1.0 | 0.5 | 0.67 |
+| Total | 10 | 6 | 4 | 4 | - | - | - |
+
+
+#### Macro Average
+
+**Macro average** is perhaps the most straightforward among the numerous options and is computed by taking an
+**unweighted** mean of all the per-class scores:
+
+$$
+\begin{align}
+\text{F1}_\text{macro} &= \frac{\text{F1}_\texttt{Airplane} + \text{F1}_\texttt{Boat} + \text{F1}_\texttt{Car}}{3} \\[1em]
+&= \frac{0.67 + 0.4 + 0.67}{3} \\[1em]
+&= 0.58
+\end{align}
+$$
+
+#### Micro Average
+
+In contrast to macro, **micro average** computes a **global** average by counting the sums of true positive (TP), false
+negative (FN) and false positive (FP):
+
+$$
+\begin{align}
+\text{F1}_\text{micro} &= \text{TODO} \\[1em]
+&= \frac{6}{6 + 0.5 \times (4 + 4)} \\[1em]
+&= 0.6
+\end{align}
+$$
+
+But what about **micro precision** and **micro recall**? These are computed with the standard precision and recall
+formulas, using the total TP/FP/FN counts across all classes:
+
+<div class="grid" markdown>
+$$
+\begin{align}
+\text{Precision}_\text{micro} &= \frac{\text{TP}_\texttt{total}}{\text{TP}_\texttt{total} + \text{FP}_\texttt{total}} \\[1em]
+&= \frac{6}{6 + 4} \\[1em]
+&= 0.6
+\end{align}
+$$
+
+$$
+\begin{align}
+\text{Recall}_\text{micro} &= \frac{\text{TP}_\texttt{total}}{\text{TP}_\texttt{total} + \text{FN}_\texttt{total}} \\[1em]
+&= \frac{6}{6 + 4} \\[1em]
+&= 0.6
+\end{align}
+$$
+</div>
+
+Note that precision, recall, and F1-score all have the same value: $0.6$. This is because micro-averaging essentially
+computes the proportion of correctly classified instances out of all instances, which is the definition of overall
+**accuracy**.
+
+In the multi-class classification cases where each sample has a single label, we get the following:
+
+$$
+\text{F1}_\text{micro} = \text{Precision}_\text{micro} = \text{Recall}_\text{micro} = \text{Accuracy}
+$$
+
+### Weighted Average
+
+**Weighted average** computes the mean of all per-class scores while considering each class’s **support**. In this case,
+support is the number of actual instances of the class in the dataset.
+
+For example, if there are 5 samples of class `Airplane`, then the support value of class `Airplane` is 5. In other
+words, support is the sum of true positive (TP) and false negative (FN) counts. The weight is the proportion of each
+class’s support relative to the sum of all support values:
+
+$$
+\begin{align}
+\text{F1}_\text{weighted} &= \left( \text{F1}_\texttt{Airplane} \times \frac{\text{#}\ \texttt{Airplane}}{\text{# Total}} \right)
```

Comment:
Thoughts on using subscript for #s? 
<img width="774" alt="image" src="https://github.com/kolenaIO/kolena/assets/17771952/5b1313c7-7320-4beb-94ac-fb0f6f8dc1f6">

Reply:
Used `\tfrac` instead and am happy with the results:

<img width="784" alt="Screen Shot 2023-06-16 at 1 24 01 PM" src="https://github.com/kolenaIO/kolena/assets/9648886/7dc739b0-2e99-46b7-81e1-cfd7968e0462">

Thanks for the nudge.